resources/providers/gcplib/inventory/provider.go

```diff
@@ -72,64 +92,94 @@ type ProviderInitializerAPI interface {
 		ListAssets: func(ctx context.Context, req *assetpb.ListAssetsRequest, opts ...gax.CallOption) Iterator {
 			return client.ListAssets(ctx, req, opts...)
 		},
 	}
 
+	// initialize GCP resource manager client
+	var gcpClientOpt []option.ClientOption
+	gcpClientOpt = append(append(gcpClientOpt, option.WithScopes(cloudresourcemanager.CloudPlatformReadOnlyScope)), gcpConfig.ClientOpts...)
+	crmService, err := cloudresourcemanager.NewService(ctx, gcpClientOpt...)
+	if err != nil {
+		return nil, err
+	}
+	// wrap the resource manager client for mocking
+	crmServiceWrapper := &ResourceManagerWrapper{
+		getProjectDisplayName: func(ctx context.Context, parent string) string {
+			prj, err := crmService.Projects.Get(parent).Context(ctx).Do()
+			if err != nil {
+				log.Errorf("error fetching GCP Project: %s", err)
+				return ""
+			}
+			return prj.DisplayName
+		},
+		getOrganizationDisplayName: func(ctx context.Context, parent string) string {
+			org, err := crmService.Organizations.Get(parent).Context(ctx).Do()
+			if err != nil {
+				log.Errorf("error fetching GCP Org: %s", err)
+				return ""
+			}
+			return org.DisplayName
+		},
+	}
+
 	return &Provider{
-		Config: gcpConfig,
-		client: wrapper,
-		log:    log,
-		ctx:    ctx,
+		Config:    gcpConfig,
+		log:       log,
+		ctx:       ctx,
+		inventory: assetsInventoryWrapper,
+		crm:       crmServiceWrapper,
 	}, nil
 }
 
-func (p *Provider) ListAllAssetTypesByName(assets []string) ([]*assetpb.Asset, error) {
-	p.log.Infof("Listing GCP assets: %v", assets)
+func (p *Provider) ListAllAssetTypesByName(assetTypes []string) ([]*ExtendedGcpAsset, error) {
+	p.log.Infof("Listing GCP asset types: %v in %v", assetTypes, p.Config.Parent)
 
 	wg := sync.WaitGroup{}
-	scope := fmt.Sprintf("projects/%s", p.Config.ProjectId)
 	var resourceAssets []*assetpb.Asset
 	var policyAssets []*assetpb.Asset
 
 	wg.Add(1)
 	go func() {
 		request := &assetpb.ListAssetsRequest{
-			Parent:      scope,
-			AssetTypes:  assets,
+			Parent:      p.Config.Parent,
+			AssetTypes:  assetTypes,
 			ContentType: assetpb.ContentType_RESOURCE,
 		}
-		resourceAssets = getAllAssets(p.log, p.client.ListAssets(p.ctx, request))
+		resourceAssets = getAllAssets(p.log, p.inventory.ListAssets(p.ctx, request))
 		wg.Done()
 	}()
 
 	wg.Add(1)
 	go func() {
 		request := &assetpb.ListAssetsRequest{
-			Parent:      scope,
-			AssetTypes:  assets,
+			Parent:      p.Config.Parent,
+			AssetTypes:  assetTypes,
 			ContentType: assetpb.ContentType_IAM_POLICY,
 		}
-		policyAssets = getAllAssets(p.log, p.client.ListAssets(p.ctx, request))
+		policyAssets = getAllAssets(p.log, p.inventory.ListAssets(p.ctx, request))
 		wg.Done()
 	}()
 
 	wg.Wait()
-
-	return mergeAssetContentType(append(resourceAssets, policyAssets...)), nil
+	var assets []*assetpb.Asset
+	assets = append(append(assets, resourceAssets...), policyAssets...)
+	mergedAssets := mergeAssetContentType(assets)
+	extendedAssets := extendAssets(p.ctx, p.crm, mergedAssets)
```

Comment:
this is where we merge assets of the same name (with diff asset type) and then extend them with display names